.github/workflows/linux.yml

```diff
@@ -0,0 +1,101 @@
+name: Linux CI
+
+on: [push, pull_request]
+
+env:
+  TRAVIS_OS_NAME: linux
+  SPEC_SPLIT_DOTS: 160
+
+jobs:
+  test_linux:
+    env:
+      ARCH: x86_64
+      ARCH_CMD: linux64
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Crystal source
+        uses: actions/checkout@v2
+
+      - name: Prepare System
+        run: bin/ci prepare_system
+
+      - name: Prepare Build
+        run: bin/ci prepare_build
+
+      - name: Test
+        run: bin/ci build
+
+  test_linux_32:
+    env:
+      ARCH: i386
+      ARCH_CMD: linux32
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Crystal source
+        uses: actions/checkout@v2
+
+      - name: Prepare System
+        run: bin/ci prepare_system
+
+      - name: Prepare Build
+        run: bin/ci prepare_build
+
+      - name: Test
+        run: bin/ci with_build_env 'make std_spec threads=1'
```

Comment:
What do you think of trying to run the compiler specs here too?

Reply:
The compiler specs are executed later using the compiled compiler. But it takes a lot of time, so this is just a "fail fast" step to catch more common issues before that.

Reply:
Note I'm talking about `test_linux_32` specifically here. I don't see any other 32bit tests here?

Reply:
Oh, sorry! Yup, that's probably a good idea. I just transferred as literal as possible what's currently running on CircleCI, but I don't know why it's not running the whole suite with `bin/ci build`. @bcardiff do you know why? Resource limits maybe?

Reply:
The reason was that the resources were not enough to build the compiler in 32 bits environment. I doubt they will be able to succeed. 

Ref: https://github.com/crystal-lang/crystal/pull/8283

Reply:
I mean otoh,  while things  like 32bit ARM is still widely used, 32bit  x86 is basically dead anyways. I guess we wouldn't be  too terrible people if we just drop it as a platform when  most Linux distributions  and OSes already did so.

Reply:
When we have some other 32 bits platform, I will accept to drop 32bits x86 builds. If we drop it before I fear there will be more things broken on 32 bits due to 64 bit assumptions.

Maybe what we can do is to drop the 32 bits compiler and use cross-compilation to test 32 bits std-spec. If we do things that way it will be less things to maintain in the release process. And at the same time pave the way for something between tier 1 and tier 2 (or improve what is offered in tier 2).

Reply:
I pocked a bit at 32bit ARVMv7 support in the last couple of days and the current state is... not pretty.

* Stdlib specs break on some random OpenSSL session handshake rejects that sound more like memory corruption  than anything else, or perhaps another round of C ABI issues.
* `Atomic` requires  some more from `compiler-rt` on the platform and just linking with `clang -rtlib=compiler-rt` had me run into all kinds of linker issues.
* `compiler_spec` just runs out of memory no matter what I do, that is not actual memory but what it can allocate via bdwgc, regardless of `--enable-large-config`.
* Running an armhf docker container on an aarch64 host was so broken, not even `date` could do anything but display random values.
* My attempt to `virt-install --arch=armv7l` on the aarch64 host would just never boot.

So, that's some road ahead of us and will likely be in the Windows kind of state for a while where we just can run a whatever works right now CI.

Reply:
bumpy road ahead 😞 .

Let's decouple the discussion of better 32 bits infra for after this PR then. Do you agree @jhass?

Reply:
Sounds good.